### DISCUSSION TITLE
Revamping SageTV

### DISCUSSION THREAD

**Discussion by techiegirl2day** [OpenSageTV/Community#56] [Developer discussions]
I am wondering what it would take to modernize this program to bring it in line with the cool kids?  IF we could come up with a solution for everyday users to have a DVR for streaming etc legally that is both user friendly and available cross platform aka Apple TV Android TV Roku etc would there be enough demand to justify the workload to get it done? I'm Sure there would be blowback from the content suppliers.  My idea is to have a good media aggregator that marries all the things we like with SageTV along with all the latest streaming options. Can we create a TIVO like experience 2.0?   Is this viable? or even desired?

Galaxyusrfer

**Comment by nyplayer** [comment]
There are no developers available to re-write SageTV which is what that would take. As far as I know there is 1 developer left keeping SageTV alive.

**Reply by Sagetvmaster** [reply]
Although, I’d like to think Narflex may be still fixing the odd thing if it were absolutely necessary, as he’s still a user I believe?

**Reply by eelpout** [reply]
Josh, jvl711, who maintained the MiniClient and an updated transcoder, was working on a new interface I think. But he abrubtly disappeared around the end of last year and hasn't been heard from since. Hope he's okay.

**Reply by Sagetvmaster** [reply]
Yes, his disappearance is a little odd.  I too hope he's ok, as he put a lot of work into the improved MiniClient.  Hopefully he will chime in and tell us that he's still around :)

**Reply by Gary0201** [reply]
Does this mean SageTV as a thing is basically dead?  That's sad.  I've really enjoyed using this and have 3 hd300 devices and a huge library of recorded stuff.  If everyone is leaving, what are they switching to?

**Reply by valnar1** [reply]
I am using Plex + Roku for 95% of my needs, except recorded shows which I still use with Sage.  Although Plex does offer a DVR service, it's not quite as good.  Honestly what keeps me using Sage is the HD300 since it has native MPEG2 support, plus FF/RW and comskip are the best in the biz.  If I could find a Plex client that was as good, I might switch.  I cancelled extended cableTV channels earlier this summer so I sold my HDPVR's and no longer need USB-UIRT, etc.

Another benefit of the Sigma SMP8654 chip in the HD300 is their native support for imaged based subtitles, like Vobsub.  I tend to watch any DVD rip with Sage instead of Plex because it won't need to transcode.

**Reply by Gary0201** [reply]
I'm completely invested in Sage w/3 HD300s and more than 3 months of recorded video (if watched 24x7).  I use PlayOn to record streamed stuff, then move that into a Sage media directory where it gets picked up, then BMT to import those as a Sage Recording so they show up in either the Recordings or Movies section.  

I agree, the extenders are wonderful and I've no idea what I would need to do if I had to outright move to something else to continue to provide that multiple location viewing that I enjoy today.

**Reply by Gary0201** [reply]
I dread leaving Sage, but after a day of research, I'm thinking a MediaPortal back-end and Kodi client/front-ends running on maybe a Pi.  Both are active open source alternatives.  Sage had a wonderful run over the past decade/two, but like all things, change happens and we have to move on.

**Comment by TGsell1** [comment]
What features does that give you vs Sage? As long as I have cable card support the only thing missing for me is closed captions on cable card IP captured content.   I don’t need support for a static setup.  Sent from my iPhoneOn Aug 5, 2023, at 8:14 AM, Gary0201 ***@***.***> wrote:﻿
I dread leaving Sage, but after a day of research, I'm thinking a MediaPortal back-end and Kodi client/front-ends running on maybe a Pi.  Both are active open source alternatives.  Sage had a wonderful run over the past decade/two, but like all things, change happens and we have to move on.

**Reply by Gary0201** [reply]
There really isn't anything more that I need out of SageTV.  It works precisely the way I like it, though I would like to see a few fixes in BMT.  The real concern I have is that if something breaks, and everyone has abandoned it, then I'm stuck.  

I'm heavily invested in it with SageTV in a VM using HDHomeRun tuners for OTA, 3 hd300 units around the house, then I supplement OTA using PlayOn within another VM and move recorded (streamed) content from Netflix, Hulu, PBS, Tubi, etc. into a SageTV video folder and BMT to set those imported recordings as recorded media so it all shows up as though they were recorded.   I've got a bit over 3 months of recorded content (if watched 24x7).  So if the thing breaks and there's nobody around to fix anything, like I said, I'm stuck.

I've on/off been looking at alternatives, and absolutely none of them are anywhere near ready to replace Sage.  MythTV is simply brutal to setup unless you're a Linux Wiz, and years ago MythUbuntu was ended.  MediaPortal 1 and 2 I never really could get working.  Then assuming a functional back end, then having to find a front-end that isn't all flash and no substance like Kodi (SageTV excels at a simple front-end interface that just works well).

**Comment by Sagetvmaster** [comment]
I haven't found a better, easier, more family-friendly way to record from satellite and cable boxes than Sage's tuner-based system.  I'm using a system that the author's sold 13 years ago.  How many software products can you say that about...

**Reply by nyplayer** [reply]
Cable Boxes what are those? just sent back my last Box. I record From my XFINITY/Comcast APP from a FireTV Stick using HDMI.

**Reply by Sagetvmaster** [reply]
5.1 DD audio too?

**Reply by eelpout** [reply]
> Cable Boxes what are those? just sent back my last Box. I record From my XFINITY/Comcast APP from a FireTV Stick using HDMI.

do you have to trigger this manually? and what about channel tuning?

**Reply by LoisGNS** [reply]
Wondering about this too. I have a HDMI-to-USB thing that I can use to record from FireStick, Roku, etc., on my Xfinity stream app (when Sage misfires or I decide I need a show after it aired) but only manually. I also use PlayOn or Audials for such cases. If anyone knows of a more automated way, please chime in!

**Reply by eelpout** [reply]
you can get 5.1 audio via an HDHomeRun Prime and via an HD-PVR with the optical audio input from a DirecTV box. but the latter for me intermittently has audio sync issues and dropouts so I turned Dolby Digital off and go with matrixed 2 channel.

**Comment by TGsell1** [comment]
Completely agree.   If you just treat the servers and support minipcs as
frozen boxes w no updating Sage should work ‘forever’ without support.
The biggest risk is the potential demise of cable card (for my setup).  If
you want improvements above and beyond the current Sage system I’m
wondering what people’s really need? I’d vote for closed captions for
 content where it’s currently
unavailable.

**Reply by eelpout** [reply]
the problem is stable playback. If you have an HD200/300 it works for most Prime and HD-PVR recorded content, aside from HVEC/x.265 of course. the Android MiniClient in its current form is unstable, at least for me. it has a real problem with raw transport streams. even Plex w/ExoPlayer can choke on these on occasion. plus, the timeline pointer still gets reset to the beginning on occasion with playback; Josh/jvl711 never did figure out the cause.

also, subtitles/closed captioning is broken as you said with the MiniClient too as far as I can tell. though I swear I got it to work once with a GTV w/Chromecast, but can't repeat it.

**Comment by TGsell1** [comment]
Thanks for reply – I guess I am only looking at things from my happy setup... I would be willing to pay for further development for sure.

**Comment by TGsell1** [comment]
I have a similar setup but using cablecard to DVR premium content and besides HD300s i use firesticks.    Sounds like I can work out dvring Netflix etc the way you did if cable card goes away and keep Sage working.   It’s really hard to complain about it.  Sent from my iPhoneOn Aug 13, 2023, at 1:41 PM, Gary0201 ***@***.***> wrote:﻿
There really isn't anything more that I need out of SageTV.  It works precisely the way I like it, though I would like to see a few fixes in BMT.  The real concern I have is that if something breaks, and everyone has abandoned it, then I'm stuck.
I'm heavily invested in it with SageTV in a VM using HDHomeRun tuners for OTA, 3 hd300 units around the house, then I supplement OTA using PlayOn within another VM and move recorded (streamed) content from Netflix, Hulu, PBS, Tubi, etc. into a SageTV video folder and BMT to set those imported recordings as recorded media so it all shows up as though they were recorded.   I've got a bit over 3 months of recorded content (if watched 24x7).  So if the thing breaks and there's nobody around to fix anything, like I said, I'm stuck.
I've on/off been looking at alternatives, and absolutely none of them are anywhere near ready to replace Sage.  MythTV is simply brutal to setup unless you're a Linux Wiz, and years ago MythUbuntu was ended.  MediaPortal 1 and 2 I never really could get working.  Then assuming a functional back end, then having to find a front-end that isn't all flash and no substance like Kodi (SageTV excels at a simple front-end interface that just works well).
